lib/ContextExclusionPlugin.js

```diff
@@ -1,12 +1,25 @@
 "use strict";
 
+/** @typedef {import("./Compiler")} Compiler */
+/** @typedef {import("./ContextModuleFactory")} ContextModuleFactory */
+
 class ContextExclusionPlugin {
+	/**
+	 * @param {RegExp} negativeMatcher Matcher regular expression
+	 */
 	constructor(negativeMatcher) {
 		this.negativeMatcher = negativeMatcher;
 	}
 
+	/**
+	 * Apply the plugin
+	 * @param {Compiler} compiler Webpack Compiler
+	 * @returns {void}
+	 */
 	apply(compiler) {
-		compiler.hooks.contextModuleFactory.tap("ContextExclusionPlugin", cmf => {
+		compiler.hooks.contextModuleFactory.tap("ContextExclusionPlugin", (
+			/** @type {ContextModuleFactory} */ cmf
```

Comment:
Why is it required? Should't `compiler.hooks.contextModuleFactory` be typed?

Reply:
Good question. Let me check

Reply:
Should not be needed